### PR TITLE
Add FindIpopt.cmake module

### PR DESCRIPTION
**Pull request by ManifoldFR**
This PR:
- adds a FindIpopt.cmake module, which enables the use of `find_package(Ipopt)`

### CODE REVIEW

File: find-external/Ipopt/FindIpopt.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 2024 CNRS INRIA
+#
+# Author: Wilson Jallet
+#
+# Adapted from: https://github.com/casadi/casadi/blob/main/cmake/FindIPOPT.cmake
```
Comment (nim65s) [inline find-external/Ipopt/FindIpopt.cmake:5]: I think this file should keep the original LGPL-3.0 license then

**Review by nim65s — APPROVED** [review]

**Comment by ManifoldFR** [comment]
Hi folks, are we merging this after all ? Do you have feedback @jorisv ?

**Review by jorisv — CHANGES_REQUESTED** [review]
I will test FindIpopt on Linux and Windows before approving

File: find-external/Ipopt/FindIpopt.cmake [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  message(STATUS "  Ipopt library found at ${ipopt_lib_path}")
+  set_target_properties(
+    ipopt
+    PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${Ipopt_INCLUDE_DIRS}"
+               INTERFACE_COMPILE_OPTIONS "${Ipopt_CFLAGS_OTHER}"
+               IMPORTED_CONFIGURATIONS "RELEASE")
+
+  if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    target_compile_definitions(ipopt INTERFACE HAVE_CSTDDEF)
+  endif()
```
Comment (jorisv) [inline find-external/Ipopt/FindIpopt.cmake:45]: I don't know if this workaround is still needed : https://github.com/ghorn/debian-coinor-ipopt/issues/2

I test without it and it's working on conda-forge Ipopt.
Maybe it's mandatory for older Ipopt.

Comment (ManifoldFR) [reply find-external/Ipopt/FindIpopt.cmake:45]: Okay, perhaps we could remove it and ask for a given minimum version if Ipopt?

**Review by jorisv — APPROVED** [review]

**Pull request merged**